yammbs/_store.py

```diff
@@ -542,24 +542,52 @@ def from_qcarchive_dataset(
 
         store = cls(database_name)
 
-        for qm_molecule in tqdm(dataset.qm_molecules, desc="Storing molecules"):
-            molecule = Molecule.from_mapped_smiles(qm_molecule.mapped_smiles, allow_undefined_stereo=True)
-            molecule.add_conformer(Quantity(qm_molecule.coordinates, "angstrom"))
+        # adapted from MoleculeRecord.from_molecule, MoleculeStore.store, and
+        # DBSessionManager.store_molecule_record
+        with store._get_session() as db:
+            # instead of DBSessionManager._smiles_already_exists
+            seen = set(db.db.query(DBMoleculeRecord.mapped_smiles))
+            for qm_molecule in tqdm(dataset.qm_molecules, desc="Storing molecules"):
+                if qm_molecule.mapped_smiles in seen:
+                    continue
+                seen.add(qm_molecule.mapped_smiles)
+                molecule = Molecule.from_mapped_smiles(qm_molecule.mapped_smiles, allow_undefined_stereo=True)
+                db_record = DBMoleculeRecord(
+                    mapped_smiles=qm_molecule.mapped_smiles,
+                    inchi_key=molecule.to_inchi(fixed_hydrogens=True),
+                )
+                db.db.add(db_record)
+                db.db.commit()
```

Comment:
Curious why this is here but there isn't the same call around L590?

Reply:
I don't quite remember, and `git log` (or `blame`) isn't helping me find it. I vaguely remember there being some issue with IDs not incrementing, but that would seem to affect the call below too, as you point out. I think that had more to do with closing and reopening the session, as the comment says. Removing the extra `commit` call doesn't cause any tests to fail, so maybe it was just overly cautious.

I'll go ahead and push that commit to avoid that confusion in the future, if that sounds good to you.

(This is mostly unrelated to this PR, but I'd be happy to delete the `CachedResultCollection` stuff too now that YDS is updated. I accidentally deleted the same line in that constructor first since they look so similar)

Reply:
> I'd be happy to delete the CachedResultCollection stuff 

Please do! I keep forgetting to put that request to text

Feel free to do that here or in a follow-up

Reply:
And re: `commit()` I leaned towards thinking it was harmless, was just confused that it wasn't in both places

Reply:
I agree it was probably pretty harmless, but calling it each time likely slows things down a bit. I'm also realizing that it was probably my first attempt to fix the session issue that I never went back and cleaned up after I found the real fix. Since I couldn't come up with a reason to comment why it was there, and the tests pass without it, I went ahead and deleted it.

Do you have any preference on deleting the `CachedResultCollection` here or in a separate PR? Separate seems cleanest, but I'm happy to do it here if that's easier. Doing it here is marginally easier for me anyway.